contribs/documentation.py

```diff
@@ -117,20 +125,24 @@ def recurse(
                     return recurse(subtype, *args, **options)
 
                 elif origin_type is Literal:
-                    return AsyncAPITypes.string(*args)
+                    if len(args) > 0:
+                        return recurse(type(subtype), enum=[subtype, *args])
+                    return recurse(type(subtype), const=subtype)
 
                 # i.e: Union[X, Y] or X | Y
+                # Note: we currently do not support multiple type in the API,
+                #       other types will be omitted from the spec
                 elif origin_type in (UnionType, Union):
                     nullable = any([item is NoneType for item in args])
-                    return recurse(subtype, *args, nullable=nullable)
```

Comment:
This is the actual fix to prevent the yaml to generate an invalid field in the spec when using unions with None